### PR TITLE
[#78] Local LLM: llama.cpp

### LINKED ISSUE #78: Local LLM: llama.cpp
Document and implement https://github.com/ggml-org/llama.cpp support

### PR DESCRIPTION
**Pull request by Sakshii-27**
Closes #78 

* Implemented and documented llama.cpp support 
*  Added unit tests in [llama_cpp.rs](cci:7://file:///Users/sakshii/pgmoneta_mcp/src/llm/llama_cpp.rs:0:0-0:0) covering text response parsing, tool call deserialization, and edge cases like empty arguments.

### CODE REVIEW

File: doc/manual/en/22-llama-cpp.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+\newpage
+
+# llama.cpp
```
[the remark this answers is not in the document]
Comment (Sakshii-27) [reply doc/manual/en/22-llama-cpp.md]: okay so currently we have 
20-llm.md
21-ollama.md 
22-llama-cpp.md(The new llama.cpp I just added)
should I merge all three into a single file (20-llm.md)? I want to make sure before I delete any existing documentation.

**Comment by jesperpedersen** [comment]
@Sakshii-27 The LLM model description can be in the "general" section

File: doc/manual/en/79-mcp-api.md [hunk 2]
```diff
@@ -17,30 +17,30 @@ The MCP implementation is built on top of the [rmcp](https://docs.rs/rmcp/latest
...
-│ PgmonetaClient  │ ◄─── Manages communication
-│   (client.rs)   │      with pgmoneta server
-└────────┬────────┘
-         │ TCP + SCRAM-SHA-256
-         │
-         ▼
-┌─────────────────┐
-│ pgmoneta server │ ◄─── Backup/restore operations
-└─────────────────┘
+``` text
```
Comment (jesperpedersen) [inline doc/manual/en/79-mcp-api.md]: The focus is on our client
